d2/src/main/java/com/linkedin/d2/discovery/stores/zk/ZooKeeperEphemeralStore.java

```diff
@@ -491,13 +491,14 @@ private void setRawClientTrackingNode()
         return;
       }
 
-      List<String> childNodes = _zk.getChildren(D2Utils.getRawClientTrackingBasePath(), false);
+      Stat nodeStat = _zk.exists(D2Utils.getRawClientTrackingBasePath(), false);
```

Comment:
actually the base path could be either /d2/..., or /shared/..., or /kafka/..., varies with the fabric. We need to the path existence one by one.

btw, what value will nodeStat be if the path not exists? also what value will nodeStat.getNumChildren() be in that case?